src/viam/trajex/totg/trajectory.cpp

```diff
@@ -1335,6 +1486,10 @@ trajectory trajectory::create(class path p, options opt, integration_points poin
                     // Compute candidate next point via Euler integration with negative dt and minimum acceleration.
                     // Negative dt reverses time direction, reconstructing velocities that led to current point.
                     // With s_ddot_min < 0 and dt < 0, s_dot increases (up) while s decreases (left).
+                    //
+                    // TODO(RSDK-12981): There's no guarantee that the candidate we select here by going
```

Comment:
This is huge and a complicated thing to fix. Effectively, we need implicit integration / solving to figure out the point we came from. I put some thoughts on how to do about it in the ticket.

Reply:
Yup I did a bunch of research on this too, we should talk it over at the whiteboard soon. But I think practically for the short term we are just going to go with the known wrong thing until we have bigger items tackled. This is a nice polish item for later, along with using a better forward explicit integrator.